src/bsp/tls/crypto-algorithms/xi_bsp_sha256.c

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <xi_bsp_crypt.h>
+#include <xi_bsp_mem.h>
+#include <sha256.h>
+
+int xi_bsp_crypt_sha256_init( void** sha )
+{
+    *sha = xi_bsp_mem_alloc( sizeof( SHA256_CTX ) );
+
+    sha256_init( ( SHA256_CTX* )*sha );
+
+    return 0;
```

Comment:
1: I think the SHA BSP module will be deleted. It will be the application's responsibility to provide a fingerprint service to a file. Although the Xively C Client may provide a "default" implementation if one doesn't want to spend much time on this, especially in case of a tutorial. Or an SHA256 may be enough for 90% of customers. So we should provide a default fingerprint service somehow, maybe no in the BSP module format. This will turn out during the fingerprint "redesign" since until now I thought it will be always SHA256.
There will be a BSP fingerprint module which the application can implement and this implementation can call back to a new Xively C Client API: xi_sha256. Which has implementations internally, pointing to wolfSSL / mbedTLS / Brad Conte's sha256 impl.
2. Unfortunately Brad Conte's functions return void.